allennlp/training/trainer.py

```diff
@@ -584,7 +590,7 @@ def _restore_checkpoint(self) -> int:
         if batch_num_total is not None:
             self._batch_num_total = batch_num_total
 
-        return epoch_to_return
+        return epoch_to_return, self._metric_tracker.best_epoch, self._metric_tracker.best_epoch_metrics
```

Comment:
what I'm saying is that the trainer can get the best epoch whenever it wants by calling `self._metric_tracker.best_epoch`, so there's no need for this method to return it.